tinyagentos/taos_agent_runtime.py

```
@@ -52,30 +52,49 @@ async def ensure_taos_opencode_server(app_state, model: str) -> OpenCodeServer:
+        # (re-scoping it to the current permitted set), else mint once and persist
+        # it. create_agent_key uses a fixed alias, so re-minting would 400 on the
+        # alias collision — persisting the value avoids that and keeps it stable.
         llm_proxy = getattr(app_state, "llm_proxy", None)
         litellm_key: str | None = None
-        if llm_proxy is not None:
+        if stored_key:
+            litellm_key = stored_key
+            if llm_proxy is not None:
+                try:
+                    await llm_proxy.update_agent_key(stored_key, permitted_models)
+                except Exception:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)